src/Engine/Renderer/RenderTechnique/RenderTechnique.cpp

```diff
@@ -1,114 +1,127 @@
-#include <Engine/Renderer/Material/BlinnPhongMaterial.hpp>
 #include <Engine/Renderer/RenderTechnique/RenderTechnique.hpp>
+
+#include <Engine/Renderer/Material/BlinnPhongMaterial.hpp>
+#include <Engine/Renderer/RenderTechnique/RenderParameters.hpp>
 #include <Engine/Renderer/RenderTechnique/ShaderConfigFactory.hpp>
 #include <Engine/Renderer/RenderTechnique/ShaderProgramManager.hpp>
 
-#include "RenderTechnique.hpp"
 #include <Core/Utils/Log.hpp>
 
 namespace Ra {
 namespace Engine {
 
 using namespace Core::Utils; // log
 
-// For iterating on the enum easily
-const std::array<RenderTechnique::PassName, 3> allPasses = {RenderTechnique::Z_PREPASS,
-                                                            RenderTechnique::LIGHTING_OPAQUE,
-                                                            RenderTechnique::LIGHTING_TRANSPARENT};
+std::shared_ptr<Ra::Engine::RenderTechnique> RadiumDefaultRenderTechnique{nullptr};
 
-std::shared_ptr<Ra::Engine::RenderTechnique> RadiumDefaultRenderTechnique( nullptr );
-
-RenderTechnique::RenderTechnique() {
-    for ( auto p : allPasses )
+RenderTechnique::RenderTechnique() : m_numActivePass{0} {
+    for ( auto p = Index( 0 ); p < m_maxNbPasses; ++p )
     {
-        shaders[p] = nullptr;
+        m_activePasses[p]     = std::move( PassConfiguration( ShaderConfiguration(), nullptr ) );
+        m_passesParameters[p] = nullptr;
     }
 }
 
 RenderTechnique::RenderTechnique( const RenderTechnique& o ) :
-    material{o.material},
-    dirtyBits{o.dirtyBits},
-    setPasses{o.setPasses} {
-    for ( auto p : allPasses )
+    m_numActivePass{o.m_numActivePass},
+    m_dirtyBits{o.m_dirtyBits},
+    m_setPasses{o.m_setPasses} {
+    for ( auto p = Index( 0 ); p < m_numActivePass; ++p )
     {
-        if ( setPasses & p )
+        if ( m_setPasses & ( 1 << p ) )
```

Comment:
Nothing prevents accessing inactive ones through `getConfiguration`

Reply:
yes but the ones from `*this` stay rather than being reinitialized

Reply:
perhaps need to update the doc (and, for belt and suspender, write an assert in `getConfiguration`) but if `hasConfiguration` return false for a pass, that will be the case after copy for un unactive pass, you don't have to use `getConfiguration` with this pass.